### ISSUE TITLE
[Bug] DLQ and BatchMessages and ConsumerInterceptor don't work well together

### ISSUE THREAD

**Issue by eolivelli** [apache/pulsar#17871]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Version

2.10.x (and possibly 2.11 and all of the other Pulsar versions), Java Client

### Minimal reproduce step

1. Send batch messages
2. Consume the messages, enabling DLQ, do not acknowledge the messages, set `batchIndexAckEnabled:true`
3. the ConsumerInterceptor#onAckTimeoutSend receives only the EntryId of the each batch (MessageIdImpl, without the batchIndex)

https://github.com/apache/pulsar/blob/99acaab22032f4614072be5d478e785a94f7e492/pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerInterceptor.java#L122

### What did you expect to see?

the DLQ processes each message in the batch independently, according to  the `batchIndexAckEnabled` policy.
When you set `batchIndexAckEnabled` you expect that every message in the batch is handled as a distinct message.

### What did you see instead?

The messages in a batch are seen as a single message.
The Set of MessageId does NEVER "contain" the MessageIds of the Messages received by the Consumer, there is no way to map them

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by lhotari** [comment]
@BewareMyPower do you have some ideas to fix this issue? I wonder if this is related to #17833 which you have been working on.

**Comment by BewareMyPower** [comment]
@lhotari No. I was just revisiting the batch index ACK implementation before. But I also noticed this issue as well. It seems like when the batch index ACK was introduced, the DLQ feature has not been considered well.
